File: algorithms/168.cpp
#include <iostream>
using namespace std;

class Solution
{
public:
    string convertToTitle(int n)
    {
        if ( n <= 26 )
            return string(1, 'A'+n-1);
        else if ( n%26 == 0 )
            return convertToTitle(n/26-1)+string(1, 'Z');
        else
            return convertToTitle(n/26)+string(1, 'A'+n%26-1);
    }
};

int main()
{
    Solution solution;
    for ( int i = 1; i <= 53; i++ )
        cout << i << "\t" << solution.convertToTitle(i) << endl;
    return 0;
}

File: algorithms/rust/168.rs
impl Solution {
    #[allow(dead_code)]
    pub fn convert_to_title(mut column_number: i32) -> String {
        let mut char_vec = vec![];
        while column_number > 0 {
            column_number -= 1;
            let ch = b'A' + (column_number % 26) as u8;
            column_number /= 26;
            char_vec.push(ch as char);
        }
        char_vec.iter().rev().collect()
    }
}

struct Solution;

fn main() {
    println!("Hello, world!");
}
